### DISCUSSION TITLE
put 2 coordinators in zigbee2mqtt

### DISCUSSION THREAD

**Discussion by sobrarbe** [Koenkk/zigbee2mqtt#17850] [General]
Hello, I wanted to know if you can put 2 coordinators.  One USB and another LAN in the same zigbee2mqtt integration that I have in a Home Assistant server.  I see people saying it's possible but I don't see any documentation on how to set it up.  all help is welcome.  Thank you.

**Comment by dsduarte** [comment]
on a zigbee network only one device can be coordinator... maybe what you are looking for is to use one of those two devices acting as router.
In order to make one device as router you need to look for flashing router firmware on it.

**Comment by deviantintegral** [comment]
You can do this, but you need two z2m instances, one for each coordinator. I did this, where two buildings each had their own zigbee network, but they used the same underlying mqtt broker and home assistant instance. It worked fine, though I needed to make sure to set a separate base topic for the second instance. For a few minutes I had them both going to the default `zigbee2mqtt/` topic which caused some panic and confusion, but luckily z2m keeps its own database separately in the config directory. Once I set the second network to something different (say `building2z2m/`) all was good!
